src/js/components/explorer/detail/visualization/treemap/ExplorerTreemap.jsx

```diff
@@ -52,6 +53,10 @@ export default class ExplorerTreemap extends React.Component {
 
     buildVirtualChart(props) {
         const data = props.data.toJS();
+
+        // remove the negative values from the data because they can't be displayed in the treemap
+        remove(data, (v) => v.amount <= 0);
```

Comment:
We need to add additional logic around handling an empty `data` set. If the full `data` array is negative or 0, then data becomes an empty array. However, by passing the empty array through the d3 `hierarchy` and `tree().leaves()` functions, we still end up with a single node in `treeItems`.

Instead of checking only if `treeItems.length` is 0, we should also check if `data.length` is 0 and set an empty `virtualChart` array in state.

Reply:
Changed the check to `if (treeItems.length === 0 || data.length === 0)`